src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatibilityAnalyzer.cs

```diff
@@ -583,6 +600,36 @@ static void RemoveOtherSupportsOnDifferentPlatforms(SmallDictionary<string, Plat
                 }
             }
 
+            static bool TryGetSymbol(ConcurrentDictionary<ISymbol, SmallDictionary<string, PlatformAttributes>?> platformSpecificMembers,
+                [NotNullWhen(true)] out ISymbol? symbol, ImmutableArray<ITypeSymbol> typeArguments)
+            {
+                foreach (var typeArgument in typeArguments)
+                {
+                    if (platformSpecificMembers.TryGetValue(typeArgument, out var foundAttributes) &&
+                        foundAttributes != null &&
+                        foundAttributes.Any())
+                    {
+                        symbol = typeArgument;
+                        return true;
```

Comment:
Hmm I am not clear on this - are we just considering the first type argument symbol that has any platform attributes? What if there are multiple type arguments with platform attributes, but the one with violation type argument is not the first one, say `if (!IsWindowsOnly()) { GenericMethod<BrowserOnlyType, WindowsOnlyType>(); }`. I believe this should still flag the method invocation.

Reply:
> are we just considering the first type argument symbol that has any platform attributes? What if there are multiple type arguments with platform attributes, but the one with violation type argument is not the first one

Yes, that is right, i will update to get the correct symbol with your example, thanks

> I believe this should still flag the method invocation.

This is flagging the method invocation, because the diagnostic attributes tied to the operations only, and some operation's symbol is not one which has platform-specific attributes. In some cases, one of the child symbols of the current symbol might be the one we want to report the diagnostic. But yes here it might give the wrong symbol when there were many platform-specific type parameters.

Reply:
> This is flagging the method invocation, because the diagnostic attributes tied to the operations only

I meant will this flag `if (IsBrowserOnly()) { WindowsOnlyMethod<BrowserOnlyType>(); }`?

Reply:
> I meant will this flag if (IsBrowserOnly()) { WindowsOnlyMethod<BrowserOnlyType>(); }?

Yes it might flag the wrong symbol, `BrowserOnlyType` instead `WindowsOnlyMethod`, for your other example: `if (IsBrowserOnly()) { GenericMethod<BrowserOnlyType, WindowsOnlyType>(); }` it might also have incorrect message `'BrowserOnlyType' is supported on 'windows'` instead `'WindowsOnlyType' is supported on 'windows'`, good catch, i will add test for both case

Reply:
When an operation symbol has associated symbols also need to be analyzed (like property or event accessors, generic type parameters) with the previous solution when a parent has diagnostic the child symbols were having repeated diagnostic and it was difficult to get the real symbol (parent or child, which child etc) having the diagnostic from operations map, refactored it to diagnose only immediate attributes for child symbols and keep operation and symbol both in the map